facebook-commerce.php

```diff
@@ -785,26 +790,38 @@ private function find_variation_product_item_ids( $product_group_id ) {
 		return $product_item_ids;
 	}
 
-
 	/**
-	 * Gets the total of published products.
+	 * Gets the total number of published products.
 	 *
 	 * @return int
 	 */
 	public function get_product_count() {
+		$product_counts = wp_count_posts( 'product' );
+		return $product_counts->publish;
+	}
 
-		$args = array(
-			'post_type'      => 'product',
-			'post_status'    => 'publish',
-			'posts_per_page' => -1,
-			'fields'         => 'ids',
+	/**
+	 * Should full batch-API sync be allowed?
+	 *
+	 * Used to disable various full sync UI/APIs to avoid performance impact.
+	 *
+	 * @return boolean True if full sync is safe.
+	 * @since 2.6.1
+	 */
+	public function allow_full_batch_api_sync() {
+		/**
+		 * Allow full batch api sync to be enabled or disabled.
+		 *
+		 * @param bool $allow Is full batch allowed based on threshold (@see WC_Facebookcommerce_Integration::MAX_PRODUCTS_FOR_FULL_SYNC)
```

Comment:
Can probably just define this as the default value for the filter. The fact that it's based on the threshold is secondary / an implementation detail.

One way to make it clear would be to move out to its own line & variable, e.g.

```php
$default_allow_full_batch_sync = ( $this->get_product_count() < self::MAX_PRODUCTS_FOR_FULL_SYNC );
```

Reply:
Updated in 211dd13